dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java

```diff
@@ -441,4 +449,36 @@ public void testNavTool_rootLevel_returnItemsOnlyForSite() throws Exception
 
     }
 
+    /**
+     * Method to test: NavTool.getNav
+     * Given scenario: get navigation for system folder ("/")
+     * Expected result: getting the navigation should not change system folder's hostId
+     * @throws Exception exception
+     */
+
+    @Test
+    public void testNavTool_getNav_SystemFolder_ShouldNotChangeSystemFolderHostId() throws Exception
+    {
+        //Get SystemFolder
+        Folder systemFolder = APILocator.getFolderAPI().findSystemFolder();
+
+        //Create new Host
+        final Host host = new SiteDataGen().nextPersisted();
+
+        //Create contentlets on one host
+        final File file = File.createTempFile("fileTestEngTrue", ".txt");
+        FileUtil.write(file, "helloworld");
+        final Contentlet fileAssetOneHost = new FileAssetDataGen(systemFolder, file)
+                .host(host).setProperty(FileAssetAPI.SHOW_ON_MENU, "true").nextPersisted();
+        final Template template = new TemplateDataGen().nextPersisted();
+        final Contentlet pageAssetOneHost = new HTMLPageDataGen(systemFolder, template)
+                .showOnMenu(true).host(host).nextPersisted();
+        ContentletDataGen.publish(pageAssetOneHost);
+        ContentletDataGen.publish(fileAssetOneHost);
+
+        new NavTool().getNav(host, systemFolder.getPath());
+        systemFolder = APILocator.getFolderAPI().findSystemFolder();
+        assertEquals(Host.SYSTEM_HOST, systemFolder.getHostId());
```

Comment:
I think we should check also the return value of the getNav method

Reply:
It's covered in a different test